nemo/collections/speechlm2/data/duplex_ear_tts_dataset.py

```diff
@@ -0,0 +1,873 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import random
+import re
+
+import torch
+import torch.nn.functional as F
+import torch.utils.data
+from lhotse import CutSet, Seconds, compute_num_frames
+from lhotse.cut import Cut
+from lhotse.dataset.collation import collate_audio, collate_vectors
+from lhotse.utils import ifnone
+
+from nemo.collections.common.tokenizers import TokenizerSpec
+from nemo.collections.speechlm2.data.utils import get_pad_id
+from nemo.collections.tts.parts.utils.helpers import get_mask_from_lengths
+from nemo.utils import logging
+
+
+def sample_audio_segments_repeat(
```

Comment:
Please move the helper functions at the bottom of this file, we should see the Dataset definition as the first thing when we open this. 

Also, since you have this nicely structured, it is now easy to test each individual operation in data prep. Can we add unit tests for them?